src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1206,19 +1218,22 @@ void primitive_inst::load(cldnn::BinaryInputBuffer& ib) {
     }
 
     _outputs[0] = nullptr;
-    if (!_mem_allocated) {
-        std::string dep_id;
-        ib >> dep_id;
-        if (dep_id.compare("NOT_FOUND") != 0) {
-            _outputs[0] = get_network().get_engine().reinterpret_buffer(get_network().get_primitive(dep_id)->output_memory(), output_layout);
-        } else if (type() == cldnn::mutable_data::type_id()) {
-            _outputs[0] = get_network().get_engine().allocate_memory(output_layout, _allocation_type);
-        }
-    } else {
-        if ((!can_share_buffer()) || can_be_optimized() || is_output()) {
-            _outputs[0] = get_network().get_engine().allocate_memory(output_layout, _allocation_type);
+    if (!output_is_null) {
+        if (!_mem_allocated) {
+            std::string dep_id;
+            ib >> dep_id;
+            if (dep_id.compare("NOT_FOUND") != 0) {
+                _outputs[0] = get_network().get_engine().reinterpret_buffer(get_network().get_primitive(dep_id)->output_memory(), output_layout);
+            } else if (type() == cldnn::mutable_data::type_id()) {
+                _outputs[0] = get_network().get_engine().allocate_memory(output_layout, _allocation_type);
+            }
```

Comment:
Now I'd change the question. When is the mem_allocated is true but output is nullptr? Should't we set mem_allocated to be false if output is nullptr?

Reply:
As I replied above, in the case of implicit concat, `mem_allocated` is false and output is not null.